### PR TITLE
Add CLI test for activation-key remove-host-collection

### LINKED ISSUE #1706: Add CLI test for activation-key remove-host-collection
```
hammer activation-key remove-host-collection --help

Usage:
    hammer activation-key remove-host-collection [OPTIONS]

Options:
 --host-collection HOST_COLLECTION_NAME  Host collection name to search by
 --host-collection-id HOST_COLLECTION_ID Id of the host collection
 --id ID                                 ID of the activation key
 --name NAME                             Activation key name to search by
 --organization ORGANIZATION_NAME        Organization name to search by
 --organization-id ORGANIZATION_ID       organization ID
 --organization-label ORGANIZATION_LABEL Organization label to search by
 -h, --help                              print help
```

### PR DESCRIPTION
**Pull request by oshtaier**
Slightly increase coverage for activation key remove host collection functionality.
Fixed previous coverage to correspond more to our code style
Closes #1706

### CODE REVIEW

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/cli/test_activationkey.py -m test_remove_host
.......
----------------------------------------------------------------------
Ran 7 tests in 245.748s

OK
```

File: tests/foreman/cli/test_activationkey.py [hunk 1]
```diff
@@ -1276,6 +1276,55 @@ def test_bugzilla_1111723(self):
 
         self.assertEqual(new_activation_key['name'], name)
 
+    def test_remove_host_by_id(self):
```
Comment (sthirugn) [inline tests/foreman/cli/test_activationkey.py]: test_remove_host_collection_by_id?

File: tests/foreman/cli/test_activationkey.py [hunk 2]
```diff
@@ -1276,6 +1276,55 @@ def test_bugzilla_1111723(self):
...
+        """@Test: Test that hosts associated to Activation Keys can be removed
+        using id of that host collection
```
Comment (sthirugn) [inline tests/foreman/cli/test_activationkey.py]: using host collection id?

File: tests/foreman/cli/test_activationkey.py [hunk 3]
```diff
@@ -1276,6 +1276,55 @@ def test_bugzilla_1111723(self):
...
+
+        @Feature: Activation key - Host
+
+        @Steps:
+
+        1. Create Activation key
+        2. Create hosts collection
```
Comment (sthirugn) [inline tests/foreman/cli/test_activationkey.py]: Create host collection

File: tests/foreman/cli/test_activationkey.py [hunk 4]
```diff
@@ -1276,6 +1276,55 @@ def test_bugzilla_1111723(self):
...
+        3. Associate the hosts to Activation key
```
Comment (sthirugn) [inline tests/foreman/cli/test_activationkey.py]: Associate the host-collection to Activation key

File: tests/foreman/cli/test_activationkey.py [hunk 5]
```diff
@@ -1276,6 +1276,55 @@ def test_bugzilla_1111723(self):
...
+        4. Remove the hosts associated to Activation key using id of that
+        collection
```
Comment (sthirugn) [inline tests/foreman/cli/test_activationkey.py]: 4. Remove the hosts associated to Activation key using host-collection id

File: tests/foreman/cli/test_activationkey.py [hunk 6]
```diff
@@ -1276,6 +1276,55 @@ def test_bugzilla_1111723(self):
...
+
+        @Assert: Hosts successfully removed from activation key
```
Comment (sthirugn) [inline tests/foreman/cli/test_activationkey.py]: Host-collection successfully removed from Activation key

File: tests/foreman/cli/test_activationkey.py (hunk 3 above)
Comment (oshtaier) [inline tests/foreman/cli/test_activationkey.py]: It is a description section, so why not use English language?
Basically, I am referring to UI and common sense

File: tests/foreman/cli/test_activationkey.py (hunk 5 above)
Comment (oshtaier) [inline tests/foreman/cli/test_activationkey.py]: It is a description section, so why not use English language?
Basically, I am referring to UI and common sense

File: tests/foreman/cli/test_activationkey.py (hunk 4 above)
Comment (oshtaier) [inline tests/foreman/cli/test_activationkey.py]: It is a description section, so why not use English language?
Basically, I am referring to UI and common sense
I can change that line to "Associate the host collection to Activation key" if you like.

File: tests/foreman/cli/test_activationkey.py [hunk 7]
```diff
@@ -1284,20 +1333,21 @@ def test_bugzilla_1111723(self):
         gen_string('utf8'),
         gen_string('html'),
     )
-    def test_remove_host(self, host_col):
+    def test_remove_host_by_name(self, host_col):
```
Comment (sthirugn) [inline tests/foreman/cli/test_activationkey.py]: test_remove_host_collection_by_name?

File: tests/foreman/cli/test_activationkey.py (hunk 6 above)
Comment (oshtaier) [inline tests/foreman/cli/test_activationkey.py]: It is a description section, so why not use English language?
Basically, I am referring to UI and common sense

File: tests/foreman/cli/test_activationkey.py [hunk 8]
```diff
@@ -1284,20 +1333,21 @@ def test_bugzilla_1111723(self):
...
         """@Test: Test that hosts associated to Activation Keys can be removed
+        using name of that host collection
```
Comment (sthirugn) [inline tests/foreman/cli/test_activationkey.py]: Please apply the suggested corrections from the previous test in here.

File: tests/foreman/cli/test_activationkey.py [hunk 9]
```diff
@@ -1284,20 +1333,21 @@ def test_bugzilla_1111723(self):
...
-        2. Create different hosts
+        2. Create hosts collection
         3. Associate the hosts to Activation key
-        4. Remove the hosts associated to Activation key
+        4. Remove the hosts associated to Activation key using name of that
+        collection
 
-        @Assert: Hosts successfully removed that
-        are associated to Activation key
+        @Assert: Hosts successfully removed from activation key
```
Comment (sthirugn) [inline tests/foreman/cli/test_activationkey.py]: Please apply the suggested corrections from the previous test in here.

File: tests/foreman/cli/test_activationkey.py (hunk 4 above)
Comment (sthirugn) [inline tests/foreman/cli/test_activationkey.py]: yes please. I am okay with using host-collection or host collection :)

File: tests/foreman/cli/test_activationkey.py [hunk 10]
```diff
@@ -1276,6 +1276,55 @@ def test_bugzilla_1111723(self):
...
+        except CLIFactoryError as err:
+            self.fail(err)
+
+        result = ActivationKey.add_host_collection({
+            u'name': activation_key['name'],
+            u'host-collection-id': new_host_col['id'],
+            u'organization': self.org['name'],
+        })
+        self.assertEqual(result.return_code, 0)
+        self.assertEqual(len(result.stderr), 0, result.stderr)
```
Comment (elyezer) [inline tests/foreman/cli/test_activationkey.py]: We already have `result.stderr` in the logs, not need to add it as the message. Please drop in other places too.

Comment (elyezer) [inline tests/foreman/cli/test_activationkey.py]: What about including:

```python
self.assertEqual(len(result.stdout['host-collections']), 1)
```

File: tests/foreman/cli/test_activationkey.py [hunk 11]
```diff
@@ -1284,20 +1333,21 @@ def test_bugzilla_1111723(self):
...
-        3. Associate the hosts to Activation key
-        4. Remove the hosts associated to Activation key
+        2. Create host collection
+        3. Associate host collection to Activation key
+        4. Remove the host collection associated to Activation key using name
+        of that collection
 
-        @Assert: Hosts successfully removed that
-        are associated to Activation key
+        @Assert: Host collection successfully removed from activation key
```
Comment (elyezer) [inline tests/foreman/cli/test_activationkey.py]: :+1:

File: tests/foreman/cli/test_activationkey.py [hunk 12]
```diff
@@ -1305,60 +1355,44 @@ def test_remove_host(self, host_col):
...
-            u'host-collection': new_host_col_name,
+            u'host-collection': new_host_col['name'],
             u'organization-id': org_id,
         })
-        self.assertEqual(
-            result.return_code, 0, 'Failed to add host-col activation key')
-        self.assertEqual(
-            len(result.stderr), 0, result.stderr)
+        self.assertEqual(result.return_code, 0)
+        self.assertEqual(len(result.stderr), 0, result.stderr)
```
Comment (elyezer) [inline tests/foreman/cli/test_activationkey.py]: Please add an assertion to check if it have 1 host-collection

File: tests/foreman/cli/test_activationkey.py [hunk 13]
```diff
@@ -1305,60 +1355,44 @@ def test_remove_host(self, host_col):
...
-            u'id': activation_key['id'],
-        })
-        self.assertEqual(
-            result.return_code, 0, 'Failed to get info for activation key')
-        self.assertEqual(
-            len(result.stderr), 0, result.stderr)
+        result = ActivationKey.info({u'id': activation_key['id']})
+        self.assertEqual(result.return_code, 0)
+        self.assertEqual(len(result.stderr), 0, result.stderr)
         self.assertEqual(
```
Comment (elyezer) [inline tests/foreman/cli/test_activationkey.py]: Please add an assertion to check if it have 1 host-collection before

File: tests/foreman/cli/test_activationkey.py (hunk 10 above)
Comment (elyezer) [inline tests/foreman/cli/test_activationkey.py]: What about grabbing the info and asserting that it have just one host-collection?

```python
self.assertEqual(len(result.stdout['host-collections']), 1)
```

**Comment by elyezer** [comment]
ACK, pending comments/

**Comment by sthirugn** [comment]
ACK pending @elyezer's comments

File: tests/foreman/cli/test_activationkey.py (hunk 10 above)
Comment (JacobCallahan) [inline tests/foreman/cli/test_activationkey.py]: :+1:

**Comment by JacobCallahan** [comment]
third ACK, pending @elyezer's suggestions (specifically dropping result.stderr)

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/cli/test_activationkey.py -m test_remove_host_collection
.......
----------------------------------------------------------------------
Ran 7 tests in 252.204s

OK
```

**Comment by Ichimonji10** [comment]
I believe all comments have been adressed. I specificially checked to make sure that `result.stderr` checks have been dropped as per @elyezer's request, and I see that that has been done.

Fourth ACK.

**Pull request merged**